lib/menubar/utils.lua

```diff
@@ -151,6 +168,68 @@ function utils.rtrim(s)
     return s
 end
 
+--- Unescape strings read from desktop files.
+-- Possible sequences are \n, \r, \s, \t, \\, and \; (only for lists) which have
+-- the usual meanings.
+-- @tparam string s String to unescape
+-- @tparam boolean list Whether the string should be escaped as a list
+-- @treturn string|table The unescaped string or a table of unescaped strings if
+-- list is true.
+function utils.unescape(s, list)
+    if not s then return end
+    -- We can't just use string.gsub because something like [[\\s]] would become
+    -- either [[ ]] or [[\ ]] (depending on the order) while the correct result
+    -- is [[\s]].
```

Comment:
How about (basically untested):
```
return string.gsub(s, "\\.", function(sequence)
    local c = sequence:sub(2, 2)
    return escape_chars[sequence:sub(2, 2)] or sequence
end)
```
The list case is a bit harder (should it get its own, separate function? this does more than just handling escape sequences). Again, some untested idea:
```
local segments = {}
local part = ""
-- Iterate over sub-strings between semicolons
for word in string.gmatch(s, "[^;]+") do
  if word:sub(-1, -1) == "\\" then
    -- The semicolon was escaped, remember this part for later
    part = part .. word:sub(1, -2) .. ";"
  else
    table.insert(segments, utils.unescape(part .. word))
    part = ""
  end
end
```